src/Microsoft.AspNet.Mvc.Core/Formatters/InputFormatter.cs

```diff
@@ -147,5 +148,41 @@ protected Encoding SelectCharacterEncoding(InputFormatterContext context)
 
             return null;
         }
+
+        /// <inheritdoc />
+        public IReadOnlyList<MediaTypeHeaderValue> GetSupportedContentTypes(MediaTypeHeaderValue contentType, Type objectType)
+        {
+            if (!CanReadType(objectType))
+            {
+                return null;
+            }
+
+            if (contentType == null)
+            {
+                // If contentType is null, then any type we support is valid.
+                return SupportedMediaTypes.Count > 0 ? new List<MediaTypeHeaderValue>(SupportedMediaTypes) : null;
+            }
+            else
+            {
+                List<MediaTypeHeaderValue> mediaTypes = null;
+
+                // Confirm this formatter supports a more specific media type than requested e.g. OK if "text/*"
```

Comment:
Sure, will update comment. I think it was copy-pasted, as was most of this code.